cli/src/main.rs

```diff
@@ -1,48 +1,32 @@
+#![allow(dead_code)]
+#![deny(warnings)]
 mod cli;
 mod repl;
 mod report;
 
-use crate::cli::Cli;
+use crate::cli::{handle_source, Cli};
 use crate::repl::prompt;
-use crate::report::FormattedError;
 use clap::Parser;
-use context::source::Source;
-use dbg_pls::color;
-use miette::GraphicalReportHandler;
-use parser::parse;
+use context::source::OwnedSource;
+use miette::MietteHandlerOpts;
 use std::io;
+use std::ops::Deref;
 use std::process::exit;
 
 fn main() -> io::Result<()> {
     let cli = Cli::parse();
 
-    let handler = GraphicalReportHandler::default();
+    miette::set_hook(Box::new(|_| {
+        Box::new(MietteHandlerOpts::new().tab_width(2).build())
+    }))
+    .expect("miette setup");
```

Comment:
```suggestion
    miette::set_hook(Box::new(|_| {
        let mut builder = MietteHandlerOpts::new();
        if matches!(env::var("TERM").as_ref().map(String::as_ref), Ok("ansi")) {
            builder = builder.color(true); // Try to detect IntelliJ
        }
        Box::new(builder.build())
    }))
    .expect("miette setup");
```

Reply:
i'll take a further look on color handlings in [`tree/cli`](https://github.com/moshell-lang/moshell/tree/cli)